redfish-core/lib/assembly.hpp

```diff
@@ -336,6 +382,97 @@ inline void handleChassisAssemblyGet(
     });
 }
 
+inline void
+    startOrStopADCSensor(const bool start,
+                         const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
+{
+    std::string method{"StartUnit"};
+    if (!start)
+    {
+        method = "StopUnit";
+    }
+
+    crow::connections::systemBus->async_method_call(
+        [asyncResp](const boost::system::error_code& ec) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR("Failed to start or stop ADCSensor:{}",
+                             ec.value());
+            messages::internalError(asyncResp->res);
+            return;
+        }
+        messages::success(asyncResp->res);
+    },
+        "org.freedesktop.systemd1", "/org/freedesktop/systemd1",
+        "org.freedesktop.systemd1.Manager", method,
+        "xyz.openbmc_project.adcsensor.service", "replace");
+}
+
+inline void afterGetDbusObjectDoBatteryCM(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& assembly, const boost::system::error_code& ec,
+    const dbus::utility::MapperGetObject& object)
+{
+    if (ec)
+    {
+        BMCWEB_LOG_DEBUG("DBUS response error");
```

Comment:
ERROR here

Reply:
Also need the ec, otherwise this trace doesn't do a lot

Reply:
Fixed.